sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -72,11 +72,19 @@ public static BatchSpanProcessorBuilder builder(SpanExporter spanExporter) {
             scheduleDelayNanos,
             maxExportBatchSize,
             exporterTimeoutNanos,
-            JcTools.newMpscArrayQueue(maxQueueSize));
+            createSpanQueue(maxQueueSize));
     Thread workerThread = new DaemonThreadFactory(WORKER_THREAD_NAME).newThread(worker);
     workerThread.start();
   }
 
+  private static Queue<ReadableSpan> createSpanQueue(int maxQueueSize) {
```

Comment:
We may use `JcTools` queue in other places, let's go ahead and add this fallback in JcTools.java

- Add try / catch to `newMpscArrayQueue`
- Add `instanceof` to `capacity` method to do appropriately

Reply:
Isn't a method called `newMpscArrayQueue` returning a `ArrayBlockingQueue` a bad thing?

Reply:
Shouldn't we introduce a new "JcTools" class that creates queues?

Reply:
It's an internal class so I'm not too worried about it. But I guess we can rename the method to `newFixedSizeQueue` though to be a bit less weird indeed.